brave/brave-core
# Fix 3650: Enable google sign in for extensions

Resolves https://github.com/brave/brave-browser/issues/3650
Resolves https://github.com/brave/brave-browser/issues/4672

Security Review: https://github.com/brave/security/issues/157 - (approved) 

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

1. Open Brave with a new profile and verify no connections are made to accounts.google.com
2. Navigate to https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi to install google-keep extension
3. Verify that login to google works with google keep.
4. Navigate to https://chrome.google.com/webstore/detail/google-calendar/gmbgaklkmjakoegficnlkhebmhkjfich to install google-calendar extension
5. Verify that login works with google calendar
6. Remove the extensions from Brave
7. Navigate to brave://settings/siteData and clear all site data
8. Navigate to brave://settings/socialBlocking
9. Disable `Allow Google login buttons on third party sites`
10. Click Relaunch
11. Install Google Keep extension and navigate to brave.com
12. Try sign-in with Google Keep extension. It should fail.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
